### ISSUE TITLE
Qwen3-4B-base训练loss不下降

### ISSUE THREAD

**Issue by white-wolf-tech** [modelscope/ms-swift#6777]
**Describe the bug**
What the bug is, and how to reproduce, better with screenshots(描述bug以及复现过程，最好有截图)
从主分支安装的ms-swift，3.11.0.dev

训练脚本

```bash
model_path=/data3/models/Qwen3-4B-Base/
PYTORCH_CUDA_ALLOC_CONF='expandable_segments:True' \
NPROC_PER_NODE=4 \
CUDA_VISIBLE_DEVICES=0,1,2,3 \
swift sft \
    --model ${model_path} \
    --train_type lora \
    --dataset 'AI-ModelScope/alpaca-gpt4-data-zh#5000' \
              'AI-ModelScope/alpaca-gpt4-data-en#5000' \
              'swift/self-cognition#5000' \
    --split_dataset_ratio 0.01 \
    --torch_dtype bfloat16 \
    --padding_free true \
    --per_device_train_batch_size 32 \
    --per_device_eval_batch_size 32 \
    --gradient_accumulation_steps 1 \
    --gradient_checkpointing true \
    --learning_rate 5e-5 \
    --lr_scheduler_type cosine \
    --lora_rank 64 \
    --lora_alpha 128 \
    --target_modules all-linear \
    --weight_decay 0.0 \
    --num_train_epochs 10 \
    --eval_steps 500 \
    --save_steps 500 \
    --save_total_limit 2 \
    --logging_steps 1 \
    --max_length 4096 \
    --output_dir output/Qwen3-14B \
    --warmup_ratio 0.03 \
    --attn_impl flash_attn \
    --save_only_model true \
    --group_by_length true \
    --deepspeed zero2 \
    --dataset_num_proc 100 \
    --dataloader_num_workers 100 \
    --loss_scale last_round_with_ignore_empty_think \
    --use_liger_kernel true
```
train loss:

<img width="675" height="451" alt="Image" src="https://github.com/user-attachments/assets/c698f0b9-9083-4ed8-907b-778b2f5e7dbb" />

eval loss:

<img width="703" height="451" alt="Image" src="https://github.com/user-attachments/assets/77791221-c66b-49f9-9850-29448a5d9600" />

**Your hardware and system info**
Write your system info like CUDA version/system/GPU/torch version here(在这里给出硬件信息和系统信息，如CUDA版本，系统，GPU型号和torch版本等)
cuda 12.8
torch 2.8.0

**Additional context**
Add any other context about the problem here(在这里补充其他信息)

**Comment by Jintao-Huang** [comment]
这个loss没问题啊

**Comment by Jintao-Huang** [comment]
loss曲线 和数据集有关的
